File: android/src/main/kotlin/com/borda/zebra_rfid_reader_sdk/utils/BordaHandheldTrigger.kt
package com.borda.zebra_rfid_reader_sdk.utils

object  BordaHandheldTrigger {
    private var mode: TriggerMode = TriggerMode.INVENTORY_PERFORM

    fun setMode(triggerMode: TriggerMode) {
        this.mode = triggerMode
    }

    fun getMode(): TriggerMode {
        return mode
    }
}

File: ios/Classes/BordaHandheldTrigger.swift
import Foundation

class BordaHandheldTrigger {
    static let shared = BordaHandheldTrigger()
    
    private var mode: TriggerMode = .inventoryPerform

    private init() {}
    
    func setMode(triggerMode: TriggerMode) {
        self.mode = triggerMode
    }
    
    func getMode() -> TriggerMode {
        return mode
    }
}

enum TriggerMode {
    case inventoryPerform
    case tagLocationingPerform
}
